src/visualization/minimap_panel.cc

```diff
@@ -247,5 +254,56 @@ void MinimapPanel::selectRange() {
   }
 }
 
+void MinimapPanel::adjustMinimaps(size_t selection_size, int grow_factor, size_t start_position) {
+  size_t full_size = sampler_->getFileOffset(sampler_->getSampleSize());
+  QVector<size_t> minimap_sizes;
+  QVector<size_t> minimap_selection_sizes;
+  QVector<size_t> minimap_start_position;
+  QVector<size_t> minimap_selection_offset;
+
+  do {
+    selection_size = std::min(full_size, selection_size);
+    minimap_selection_sizes.append(selection_size);
+    size_t minimap_size = std::min(static_cast<size_t>(selection_size * grow_factor), full_size);
```

Comment:
AFAIR we use uint64_t in other places for file pos, this won't work well on 32-bit. Please fix this also in other places in this PR.